runtime/interpreter/value.go

```diff
@@ -9337,6 +9338,78 @@ func (PathValue) IsStorable() bool {
 	return true
 }
 
+func ConvertPublicPath(value Value) Value {
+	stringValue, ok := value.(*StringValue)
+	if !ok {
+		return NilValue{}
+	}
+
+	ok = true
+	_ = sema.CheckPathLiteral(
+		"public",
+		stringValue.Str,
+		func(func(*ast.PathExpression) error) {
+			ok = false
+		},
+	)
+	if !ok {
+		return NilValue{}
+	}
+
+	return NewSomeValueOwningNonCopying(PathValue{
+		Domain:     common.PathDomainPublic,
+		Identifier: stringValue.Str,
+	})
+}
+
+func ConvertPrivatePath(value Value) Value {
+	stringValue, ok := value.(*StringValue)
```

Comment:
Maybe avoid the code duplication in `ConvertPublicPath`, `ConvertPrivatePath`, and `ConvertStoragePath`: They all share the same code, the only difference is the domain. Maybe introduce a helper function that accepts the domain as a parameter